test/commands/login.test.js

```diff
@@ -1,38 +1,108 @@
-const Nock = require('@fancy-test/nock').default;
-const { expect, test } = require('@oclif/test');
 const jwt = require('jsonwebtoken');
+const nock = require('nock');
 
-const fancy = test.register('nock', Nock);
+const testCli = require('./test-cli');
+const LoginCommand = require('../../src/commands/login');
 
 describe('login', () => {
-  describe('with trivial mail and bad token in args', () => {
-    fancy
-      .stdout({ print: true })
-      .stderr()
-      .command(['login', '-e', 'smile@gmail.com', '-t', 'bad_token'])
-      .it('should warn about bad token', (ctx) => {
-        expect(ctx.stderr).to.contain('Invalid token. Please enter your authentication token.');
+  describe('with email in args', () => {
+    describe('with bad token in args', () => {
+      it('should display invalid token', () => testCli({
+        inputs: [
+          `${jwt.sign({}, 'key', { expiresIn: '1day' })}\n`,
+        ],
+        command: () => LoginCommand.run(['-e', 'smile@gmail.com', '-t', 'bad_token']),
+        errorOutputs: [
+          'Invalid token. Please enter your authentication token.',
+        ],
+      }));
+    });
+    describe('with valid token in args', () => {
+      it('should login successful', () => testCli({
+        inputs: [
+          `${jwt.sign({}, 'key', { expiresIn: '1day' })}\n`,
+        ],
+        command: () => LoginCommand.run(['-e', 'smile@gmail.com', '-t', jwt.sign({}, 'key', { expiresIn: '1day' })]),
+        outputs: [
+          'Login successful',
+        ],
+      }));
+    });
+    describe('with a google mail', () => {
+      describe('with a valid token from input', () => {
+        it('should login successful', () => testCli({
+          env: { FOREST_URL: 'http://localhost:3001' },
+          nock: nock('http://localhost:3001')
+            .get('/api/users/google/robert@gmail.com')
+            .reply(200, { data: { isGoogleAccount: true } }),
```

Comment:
[Thinking] This looks like test environment configuration. We could maybe isolate it somewhere, as this is done with `mirage` in the front app. 

We would have one mockable server, activable or not, on which we could mock some calls.

This is just a thought and is ignorable.

Reply:
I agree. Let's build this together :)